__tests__/commands/install/integration.js

```diff
@@ -437,6 +438,27 @@ test.concurrent('install with file: protocol as default', (): Promise<void> => {
   });
 });
 
+// When local packages are installed, dependencies with different forms of the same relative path
+// should be deduped e.g. 'file:b' and 'file:./b'
+test.concurrent('install file: dedupe dependencies 1', (): Promise<void> => {
+  return runInstall({}, 'install-file-dedupe-dependencies-1', async config => {
+    // Check that b is not added as a sub-dependency of a
+    expect(await fs.exists(path.join(config.cwd, 'node_modules', 'a', 'node_modules'))).toEqual(false);
+  });
+});
+
+// When local packages are installed, dependencies with relative and absolute paths should be
+// deduped e.g. 'file:b' and 'file:/absolute/path/to/b'
+test.concurrent('install file: dedupe dependencies 2', (): Promise<void> => {
+  return runInstall({}, 'install-file-dedupe-dependencies-2', async (config, reporter) => {
+    // Add b as a dependency, using an absolute path
+    await add(config, reporter, {}, [`b@file:${path.join(config.cwd, 'b')}`]);
```

Comment:
`fs.realpath` might be better than using `path.join` with `config.cwd` since it is guaranteed to be absolute where as `config.cwd` can still be a relative path. Unlikely, but just pointing it out.